openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ListenerGenerator.java

```diff
@@ -0,0 +1,168 @@
+import io.ballerina.compiler.syntax.tree.MappingConstructorExpressionNode;
+import io.ballerina.compiler.syntax.tree.MappingFieldNode;
+import io.ballerina.compiler.syntax.tree.Minutiae;
+import io.ballerina.compiler.syntax.tree.MinutiaeList;
+import io.ballerina.compiler.syntax.tree.NamedArgumentNode;
+import io.ballerina.compiler.syntax.tree.NodeFactory;
+import io.ballerina.compiler.syntax.tree.ParenthesizedArgList;
+import io.ballerina.compiler.syntax.tree.PositionalArgumentNode;
+import io.ballerina.compiler.syntax.tree.QualifiedNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SeparatedNodeList;
+import io.ballerina.compiler.syntax.tree.SimpleNameReferenceNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.Token;
+import io.ballerina.openapi.exception.BallerinaOpenApiException;
+import io.ballerina.openapi.generators.GeneratorUtils;
+import io.swagger.v3.oas.models.servers.Server;
+import io.swagger.v3.oas.models.servers.ServerVariables;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.List;
+
+import static io.ballerina.openapi.generators.GeneratorConstants.CONFIG;
+import static io.ballerina.openapi.generators.GeneratorConstants.HOST;
+import static io.ballerina.openapi.generators.GeneratorConstants.NEW;
+
+/**
+ * This util class for processing the mapping in between openAPI server section with ballerina listeners.
+ *
+ * @since 2.0.0
+ */
+public class ListenerGenerator {
+    private static final int httpPort = 80;
+    private static final int httpsPort = 443;
+    private String basePath = "/";
+
+    public ListenerGenerator() {
+    }
+
+    public String getBasePath() {
+        return basePath;
+    }
+
+    /**
+     * Generate listener accoring to the given server details.
+     *
+     * @param servers  OAS server details
+     * @return         {@link ListenerDeclarationNode} for server.
+     * @throws BallerinaOpenApiException when process break with exception
+     */
+    public ListenerDeclarationNode getListenerDeclarationNodes(List<Server> servers) throws BallerinaOpenApiException {
+        // Assign host port value to listeners
+        String host;
+        int port;
+        Server server = servers.get(0);
+        if (server.getVariables() != null) {
+            ServerVariables variables = server.getVariables();
+            URL url;
+            try {
+                String resolvedUrl = GeneratorUtils.buildUrl(server.getUrl(), variables);
+                url = new URL(resolvedUrl);
+                host = url.getHost();
+                basePath = url.getPath();
+                port = url.getPort();
+                boolean isHttps = "https".equalsIgnoreCase(url.getProtocol());
+
+                if (isHttps) {
+                    port = port == -1 ? httpsPort : port;
+                } else {
+                    port = port == -1 ? httpPort : port;
+                }
```

Comment:
This can be simplified as below.
```suggestion
                if (port < 0) {
                    port = isHttps ? httpsPort : httpPort;
                }
```